fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/mixin/resource/loader/ReloadableResourceManagerImplMixin.java

```diff
@@ -40,7 +43,20 @@
 	private ResourceType type;
 
 	@Inject(at = @At("HEAD"), method = "beginReloadInner")
-	public void reload(Executor var1, Executor var2, List<ResourceReloadListener> listeners, CompletableFuture future, CallbackInfoReturnable<CompletableFuture> info) {
+	private void reload(Executor var1, Executor var2, List<ResourceReloadListener> listeners, CompletableFuture future, CallbackInfoReturnable<CompletableFuture> info) {
 		ResourceManagerHelperImpl.sort(type, listeners);
 	}
+
+	// private static synthetic method_29491(Ljava/util/List;)Ljava/lang/Object;
+	// Supplier lambda in beginMonitoredReload method.
+	@Inject(method = "method_29491", at = @At("HEAD"), cancellable = true)
+	private static void getResourcePackNames(List<ResourcePack> packs, CallbackInfoReturnable<String> cir) {
+		cir.setReturnValue(packs.stream().map(pack -> {
+			if (pack instanceof GroupResourcePack) {
+				return ((GroupResourcePack) pack).getFullName();
+			} else {
+				return pack.getName();
```

Comment:
Imo patching resoruce pack profile to return multi packs is easier, as resource pack itself is used in way more places